src/did/did-resolver.ts

```diff
@@ -36,7 +38,14 @@ export class DidResolver {
       throw new Error(`${didMethod} DID method not supported`);
     }
 
-    const resolutionResult = await didResolver.resolve(did);
+    const resolutionResult = await this.memoryCache.has(did) ? await this.memoryCache.get(did): await didResolver.resolve(did);
+    const cache = await this.memoryCache.has(did);
+    if (cache){
+      await this.memoryCache.set(did,resolutionResult);
```

Comment:
Looks like a bug? If result is in cache, no need to store it in cache again, you meant to do `!cache`?